### PR TITLE
Patch: fix seports to allow integers or string

### LINKED ISSUE #60968: seport: Doesn't support numerical ports
##### SUMMARY
If `seport` is given a numerical port, or list of numerical ports. It fails to parse. This is contrary to the documentation examples.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

seport

##### ANSIBLE VERSION

```paste below
ansible 2.8.4
  config file = /Users/dkimsey/.ansible.cfg
  configured module search path = ['/Users/dkimsey/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dkimsey/.virtualenvs/ansible-2.8/lib/python3.7/site-packages/ansible
  executable location = /Users/dkimsey/.virtualenvs/ansible-2.8/bin/ansible
  python version = 3.7.3 (default, May  1 2019, 10:48:04) [Clang 10.0.0 (clang-1000.11.45.5)]
```

##### CONFIGURATION

```paste below
ANSIBLE_NOCOWS(/Users/dkimsey/.ansible.cfg) = True
ANSIBLE_PIPELINING(/Users/dkimsey/.ansible.cfg) = True
ANSIBLE_SSH_ARGS(/Users/dkimsey/.ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=300s -o ConnectTimeout=30
DEFAULT_BECOME(/Users/dkimsey/.ansible.cfg) = True
DIFF_ALWAYS(/Users/dkimsey/.ansible.cfg) = True
RETRY_FILES_ENABLED(/Users/dkimsey/.ansible.cfg) = False
```

##### OS / ENVIRONMENT

OS X, Python 3.7

##### STEPS TO REPRODUCE

```yaml
- hosts: all
   tasks:
    - name: Allow memcached to listen on tcp ports 10000 and 10112
      seport:
        ports:
          - 10000
          - 10112
        proto: tcp
        setype: memcache_port_t
        state: present
```

##### EXPECTED RESULTS

It should parse & apply correctly.

##### ACTUAL RESULTS

It throws an exception, `AttributeError: 'int' object has no attribute 'split'`

```paste below
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 114, in <module>\n  File \"<stdin>\", line 106, in _ansiballz_main\n  File \"<stdin>\", line 49, in invoke_module\n  File \"/tmp/ansible_seport_payload_EvYc5a/__main__.py\", line 307, in <module>\n  File \"/tmp/ansible_seport_payload_EvYc5a/__main__.py\", line 297, in main\n  File \"/tmp/ansible_seport_payload_EvYc5a/__main__.py\", line 206, in semanage_port_add\n  File \"/tmp/ansible_seport_payload_EvYc5a/__main__.py\", line 159, in semanage_port_get_type\nAttributeError: 'int' object has no attribute 'split'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

```
Traceback (most recent call last):
  File "<stdin>", line 114, in <module>
  File "<stdin>", line 106, in _ansiballz_main
  File "<stdin>", line 49, in invoke_module
  File "/tmp/ansible_seport_payload_EvYc5a/__main__.py", line 307, in <module>
  File "/tmp/ansible_seport_payload_EvYc5a/__main__.py", line 297, in main
  File "/tmp/ansible_seport_payload_EvYc5a/__main__.py", line 206, in semanage_port_add
  File "/tmp/ansible_seport_payload_EvYc5a/__main__.py", line 159, in semanage_port_get_type
AttributeError: 'int' object has no attribute 'split'
```

### PR DESCRIPTION
**Pull request by joshland**
##### SUMMARY
I was unable to use seport with numbered ports.

Fixes #60968

Concurrently added the same check the Selinux python utils: https://github.com/SELinuxProject/selinux/pull/189

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
seport

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
```paste below
- name: Permit Nginx to listen on 8050
  seport:
    ports:
      - 8050-8052
      - 443
      - 80
    proto: tcp
    setype: http_port_t
    state: present
    reload: yes
```

<!--- Paste verbatim command output below, e.g. before and after your change -->
**pre Change**
```
{"changed": false, "module_stderr": "Shared connection to localhost closed.
", "module_stdout": "Traceback (most recent call last):
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 102, in <module>
    _ansiballz_main()
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574300490.251246-265170046283270/AnsiballZ_seport.py\", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.system.seport', init_globals=None, run_name='__main__', alter_sys=True)
  File \"/usr/lib64/python3.7/runpy.py\", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File \"/usr/lib64/python3.7/runpy.py\", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File \"/usr/lib64/python3.7/runpy.py\", line 85, in _run_code
    exec(code, run_globals)
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 311, in <module>
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 301, in main
  File \"/tmp/ansible_seport_payload_rsxr600w/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 212, in semanage_port_add
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1149, in add
    self.__add(port, proto, serange, type)
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1098, in __add
    (k, proto_d, low, high) = self.__genkey(port, proto)
  File \"/usr/lib/python3.7/site-packages/seobject.py\", line 1068, in __genkey
    ports = port.split(\"-\")
AttributeError: 'int' object has no attribute 'split'n
", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Post Change**
```paste below
task path: /home/eis/AnsibleTower-awx/ansible/roles/awx/tasks/main.yml:23                                                                                                                                                                    <localhost> ESTABLISH SSH CONNECTION FOR USER: eis                                                                                                                                                                                           <localhost> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb localhost '/bin/sh -c '"'"'echo ~eis && sleep 0'"'"''                                                                                                                          <localhost> (0, b'/home/eis\n', b'')                                                                                                                                                                                                         <localhost> ESTABLISH SSH CONNECTION FOR USER: eis                                                                                                                                                                                           <localhost> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb localhost '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746 `" && echo ansible-tmp-1574304761.8228998-127816148453746="` echo /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746 `" ) && sleep 0'"'"''                                                                                                                               <localhost> (0, b'ansible-tmp-1574304761.8228998-127816148453746=/home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746\n', b'')                                                                                              Using module file /srv/ansible/lib64/python3.7/site-packages/ansible/modules/system/seport.py                                                                                                                                                <localhost> PUT /home/eis/.ansible/tmp/ansible-local-34171kcsvluim/tmpat3imjkc TO /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/AnsiballZ_seport.py                                                                  <localhost> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb '[localhost]'                                                                                                                                                            <localhost> (0, b'sftp> put /home/eis/.ansible/tmp/ansible-local-34171kcsvluim/tmpat3imjkc /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/AnsiballZ_seport.py\n', b'')                                                <localhost> ESTABLISH SSH CONNECTION FOR USER: eis                                                                                                                                                                                           <localhost> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb localhost '/bin/sh -c '"'"'chmod u+x /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/ /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/AnsiballZ_seport.py && sleep 0'"'"''                                                                                                                                                                                                      <localhost> (0, b'', b'')                                                                                                                                                                                                                    <localhost> ESTABLISH SSH CONNECTION FOR USER: eis                                                                                                                                                                                           <localhost> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb -tt localhost '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-axqxkonfpogkpfrfgqkzhmxmnhrifbla ; /usr/bin/python /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/AnsiballZ_seport.py'"'"'"'"'"'"'"'"' && sleep 0'"'"''                                                                                                                                        Escalation succeeded                                                                                                                                                                                                                         <localhost> (0, b'\r\n{"ports": ["8050-8052", 443, 80], "proto": "tcp", "setype": "http_port_t", "state": "present", "changed": true, "invocation": {"module_args": {"ports": ["8050-8052", 443, 80], "proto": "tcp", "setype": "http_port_t", "state": "present", "reload": true, "ignore_selinux_state": false}}}\r\n', b'Shared connection to localhost closed.\r\n')                                                                                                                  <localhost> ESTABLISH SSH CONNECTION FOR USER: eis                                                                                                                                                                                           <localhost> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="eis"' -o ConnectTimeout=10 -o ControlPath=/home/eis/.ansible/cp/0e9b8a8fdb localhost '/bin/sh -c '"'"'rm -f -r /home/eis/.ansible/tmp/ansible-tmp-1574304761.8228998-127816148453746/ > /dev/null 2>&1 && sleep 0'"'"''                                   <localhost> (0, b'', b'')                                                                                                                                                                                                                    changed: [localhost] => {                                                                                                                                                                                                                        "changed": true,                                                                                                                                                                                                                             "invocation": {                                                                                                                                                                                                                                  "module_args": {                                                                                                                                                                                                                                 "ignore_selinux_state": false,                                                                                                                                                                                                               "ports": [
                "8050-8052",
                443,
                80
            ],
            "proto": "tcp",
            "reload": true,
            "setype": "http_port_t",
            "state": "present"
        }
    },
    "ports": [
        "8050-8052",
        443,
        80
    ],
    "proto": "tcp",
    "setype": "http_port_t",
    "state": "present"
}
```

### CODE REVIEW

**Comment by joshland** [comment]
I pasted the wrong bug.  I pasted the crash output from the Selinux/seobject.py

Here is the Ansible seport.py report.

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'int' object has no attribute 'split'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Shared connection to localhost closed.
", "module_stdout": "
Traceback (most recent call last):
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574296336.4021213-116184689972453/AnsiballZ_seport.py\", line 102, in <module>
    _ansiballz_main()
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574296336.4021213-116184689972453/AnsiballZ_seport.py\", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/home/eis/.ansible/tmp/ansible-tmp-1574296336.4021213-116184689972453/AnsiballZ_seport.py\", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.system.seport', init_globals=None, run_name='__main__', alter_sys=True)
  File \"/usr/lib64/python3.7/runpy.py\", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File \"/usr/lib64/python3.7/runpy.py\", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File \"/usr/lib64/python3.7/runpy.py\", line 85, in _run_code
    exec(code, run_globals)
  File \"/tmp/ansible_seport_payload_ijv27rpe/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 307, in <module>
  File \"/tmp/ansible_seport_payload_ijv27rpe/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 297, in main
  File \"/tmp/ansible_seport_payload_ijv27rpe/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 206, in semanage_port_add
  File \"/tmp/ansible_seport_payload_ijv27rpe/ansible_seport_payload.zip/ansible/modules/system/seport.py\", line 159, in semanage_port_get_type
AttributeError: 'int' object has no attribute 'split'
", "msg": "MODULE FAILURE
See stdout/stderr for the exact error", "rc": 1}                     
```

File: lib/ansible/modules/system/seport.py [hunk 1]
```diff
@@ -156,9 +156,13 @@ def semanage_port_get_type(seport, port, proto):
...
     :return: Tuple containing the SELinux type and MLS/MCS level, or None if not found.
     """
-    ports = port.split('-', 1)
-    if len(ports) == 1:
-        ports.extend(ports)
+    if isinstance(port, str):
+        ports = port.split('-', 1)
+        if len(ports) == 1:
+            ports.extend(ports)
+            pass
```
Comment (Akasurde) [inline lib/ansible/modules/system/seport.py]: `pass` seems useless to me.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/system/seport.py (hunk 1 above)
Comment (joshland) [reply lib/ansible/modules/system/seport.py]: Is that a thing?  I will remove it, I do it thoughtlessly.  It is literally irrelevant, it compiles out, and it ensures proper formatting during cut and pasting.

However, I know little/nothing about code hygiene in Ansible.

**Comment by joshland** [comment]
Do I need to submit this patch for 2.9.x as well?

**Pull request merged**